### PR TITLE
feat(skywalking): allow destory and configure report interval for rep…

### LINKED ISSUE #2895: Can't disable skywalking plugin on the fly
Currently we lack a way to disable the background report timer which sends the data to skywalking. We should provide a new method in the skywalking client to stop the background timer.

### PR DESCRIPTION
**Pull request by spacewander**
…orter

Fix #2895

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/plugins/skywalking.md [hunk 1]
```diff
@@ -92,6 +92,7 @@ We can set the endpoint by specified the configuration in `conf/config.yaml`.
 | service_name | string | "APISIX" | service name for skywalking reporter                                 |
 |service_instance_name|string|"APISIX Instance Name" | service instance name for skywalking reporter |
 | endpoint     | string | "http://127.0.0.1:12800" | the http endpoint of Skywalking, for example: http://127.0.0.1:12800 |
```
Comment (Firstsawyou) [inline docs/en/latest/plugins/skywalking.md:94]:
```suggestion
| endpoint_addr     | string | "http://127.0.0.1:12800" | the http endpoint of Skywalking, for example: http://127.0.0.1:12800 |
```
The `endpoint` field name is typo, we should fix it by the way.

File: docs/zh/latest/plugins/skywalking.md [hunk 2]
```diff
@@ -96,6 +96,7 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f1
 | service_name | string |  "APISIX" | skywalking 上报的 service 名称                                 |
 |service_instance_name|string| "APISIX Instance Name" | skywalking 上报的 service 实例名 |
 | endpoint     | string | "http://127.0.0.1:12800" | Skywalking 的 HTTP endpoint 地址，例如：http://127.0.0.1:12800 |
```
Comment (Firstsawyou) [inline docs/zh/latest/plugins/skywalking.md:98]: ditto.

**Review by Firstsawyou — CHANGES_REQUESTED** [review]

File: docs/zh/latest/plugins/skywalking.md (hunk 2 above)
Comment (spacewander) [reply docs/zh/latest/plugins/skywalking.md]: @Firstsawyou 
Updated.

**Review by tokers — APPROVED** [review]

File: apisix/plugins/skywalking.lua [hunk 3]
```diff
@@ -127,8 +130,22 @@ function _M.init()
...
+    if local_plugin_info.report_interval then
+        sk_cli.backendTimerDelay = local_plugin_info.report_interval
+    end
+
     sk_cli:startBackendTimer(local_plugin_info.endpoint_addr)
 end
 
 
+function _M.destroy()
+    if process.type() ~= "worker" and process.type() ~= "single" then
```
Comment (membphis) [inline apisix/plugins/skywalking.lua:142]: we can remove `process.type() ~= "single"`, right?

**Pull request merged**